Document kubernetes.io/kube-apiserver-serving ClusterTrustBundle signer (#48492)
kubernetes/website: content/en/docs/reference/command-line-tools-reference/feature-gates/ClusterTrustBundle.md

    defaultValue: false

>     fromVersion: "1.27"

Reviewer: Need to add beta details?

Author: Rebased to get this from the current main.

---
This feature gate exists in the Kubernetes API server and the controller manager.

> Used from the kube-apiserver, it enables ClusterTrustBundle support.

Reviewer: Need the feature gate on the kubelet to use the volume right?

Author: Yes, that's documented elsewhere. This FG only controls the API and the signer.

    defaultValue: false
    fromVersion: "1.33"
---

> This feature gate exists in the Kubernetes API server and the controller manager.
> 
> Used from the kube-apiserver, it enables ClusterTrustBundle support.
> 
> In order to use the ClusterTrustBundle API in your cluster, you need to enable this feature gate
> and also [enable](/docs/tasks/administer-cluster/enable-disable-api/) the associated alpha API group
> using the `--runtime-config` command line argument to kube-apiserver.
> 
> In the Kubernetes controller manager, it is used to control publishing of a ClusterTrustBundle
> for the `kubernetes.io/kube-apiserver-serving` signer.

Reviewer 2: We keep these descriptions short; most are one paragraph.

```suggestion
Enable use of [ClusterTrustBundles](/docs/reference/access-authn-authz/certificate-signing-requests/#cluster-trust-bundles).

If this feature gate is enabled for the kube-controller-manager, the control plane automatically publishes a ClusterTrustBundle
for the `kubernetes.io/kube-apiserver-serving` signer.

In order to use the ClusterTrustBundle API in your cluster, you need to enable this feature gate
and also [enable](/docs/tasks/administer-cluster/enable-disable-api/) the associated alpha API group.
```

We might one day render details of what components support a feature gate, but:
- if we do, that'll be autogenerated
- outside of a configuration rollout, the Kubernetes project recommends setting a given feature gate enabled or disabled across your entire cluster
